### DISCUSSION TITLE
Lesson 11: Error, contract not verifying on sepolia.etherscan

### DISCUSSION THREAD

**Discussion by mrthedude** [Cyfrin/foundry-full-course-cu#761] [General]
I'm am following on lesson 11 at [TIMESTAMP](https://youtu.be/sas02qSFZ74?t=30505) and want to try and deploy my ```DeployBasicNft``` contract like Patrick does. When I try to deploy the contact using ```make deploy ARGS="--sepolia network"```, everything deploys correct however I get an small error at the end of the terminal message:

 
![Screenshot 2023-08-30 154219](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/62ea7e26-fb85-4e8a-b830-03f7c0c675b0)

The message seems to be referring to my ```Makefile``` but I'm not sure why this would be an issue as a copied it directly from the course repo, here's a screenshot for reference though just in case 

 
![Screenshot 2023-08-30 154551](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/4749e675-5fdc-4a68-b5b8-55b1498d9d28)

My contracts seem to be fine, they pass with both ```forge compile``` and ```forge test```. I even tried to deploy to Anvil with ```make deploy``` to see if that errors but everything passed.

I checked the [Discussions ](https://github.com/Cyfrin/foundry-full-course-f23/discussions) page and the closest related issue I found was #445 but nothing changed after reinstalling and updating everything.

Any help is much appreciated! 🙏 🙏

**Comment by shikhar229169** [comment]
Check if you have defined the ETHERSCAN_API_KEY in your .env file correctly.

**Reply by mrthedude** [reply]
yeah it matches with my Alchemy key, here's a pic of the two
![Screenshot 2023-08-30 163922](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/8e6fd0a2-b31c-46d1-a1cf-271a2da76f47)

**Reply by shikhar229169** [reply]
No, you have entered the rpc url api key.
You need to enter the api key from etherscan.
Go to etherscan explorer website and create an api key from there.

**Reply by mrthedude** [reply]
omg idk why i did that hahahaah thankyou so much for your help! It's all verified on etherscan with no issues 🥳
